Sources/Application/MainCoordinator.swift

```diff
@@ -171,12 +171,33 @@ extension MainCoordinator: ParentCoordinator {
         switch child {
         case _ as LoginCoordinator:
             updateToken()
-        case _ as ProfileCoordinator:
-            fullLogin()
-            homeCoordinator?.baseVc?.isLoggedIn(false)
-            root.selectedIndex = 0
         default:
             break
         }
     }
+    
+    func performChildCleanup(child: ChildCoordinator) {
```

Comment:
More of a question than anything: has the functionality of this method been put here so we can switch on any future children and make life a bit easier? Or are there also performance/functional reasons behind moving most of the functionality from ProfileCoordinator?

Reply:
So this `performChildCleanup` method is part of the [Coordination package](https://github.com/govuk-one-login/mobile-ios-coordination/blob/main/Sources/Coordination/ParentCoordinator.swift) and is usually called as part of `childDidFinish` which is actually what `finish()` calls from the child coordinator.
childDidFinish removes the coordinator from the list of child coordinators on the parent though which isn't really what we want for our tab coordinators, so being able to call this method allows us to react to behaviour from the child on the parent. It's maybe a bit hacky but a useful way of us being able to reuse code from coordination.
As you say, we can switch on children in future if we want to do this from more tab coordinators to main or other child coordinators for any other parent coordinators 🙂